source/security/authentication.txt

```diff
@@ -0,0 +1,390 @@
+         :language: php
+         :dedent:
+         :start-after: start-mongodb-aws-uri
+         :end-before: end-mongodb-aws-uri
+
+.. _php-mongodb-aws-env-vars:
+
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~~
+
+If you don't provide a username and password when you construct your ``MongoDB\Client``
+object, the driver tries to retrieve AWS credentials from the following
+environment variables:
+
+- ``AWS_ACCESS_KEY_ID``
+- ``AWS_SECRET_ACCESS_KEY``
+- ``AWS_SESSION_TOKEN``
+
+To use these environment variables to authenticate your application, first set them to the
+AWS IAM values needed for authentication. You can run the ``export`` command in your shell or
+add the variables to a ``.env`` file, as shown in the following code example:
+
+.. tabs::
+
+   .. tab:: Shell Commands
+      :tabid: shell
+
+      .. code-block:: sh
+
+         export AWS_ACCESS_KEY_ID=<AWS IAM access key ID>
+         export AWS_SECRET_ACCESS_KEY=<AWS IAM secret access key>
+         export AWS_SESSION_TOKEN=<AWS session token>
+
+   .. tab:: .env File
+      :tabid: dotenv
+
+      .. code-block:: php
+
+         AWS_ACCESS_KEY_ID=<AWS IAM access key ID>
+         AWS_SECRET_ACCESS_KEY=<AWS IAM secret access key>
+         AWS_SESSION_TOKEN=<AWS session token>
+
+.. important::
+  
+   Don't percent-encode the values in these environment variables.
+
+After you set these environment variables, set the ``authMechanism``
+connection option to ``'MONGODB-AWS'``.
+
+.. _php-mongodb-aws-env-example:
+
+Example
+```````
+
+The following example sets the ``authMechanism`` connection option. 
+You can set this option in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through a parameter in your connection URI.
+
+.. tabs::
+
+   .. tab:: MongoDB\\Client
+      :tabid: Client
+
+      .. literalinclude:: /includes/authentication.php
+         :language: php
+         :dedent:
+         :start-after: start-mongodb-aws-env-client
+         :end-before: end-mongodb-aws-env-client
+
+   .. tab:: Connection URI
+      :tabid: connectionstring
+
+      .. literalinclude:: /includes/authentication.php
+         :language: php
+         :dedent:
+         :start-after: start-mongodb-aws-env-uri
+         :end-before: end-mongodb-aws-env-uri
+
+.. tip:: AWS Lambda
+   
+   AWS Lambda runtimes can automatically set these environment variables during
+   initialization. For more information about using environment variables in an AWS Lambda
+   environment, see `Using Lambda environment variables
+   <https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime>`__
+   in the AWS documentation.
+
+.. _php-mongodb-aws-oidc:
+
+AssumeRoleWithWebIdentity Request
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If your application authenticates users for your EKS cluster from an OpenID Connect (OIDC)
+identity provider, the C driver can make an ``AssumeRoleWithWebIdentity`` request
```

Comment:
Here I would just refer to "the driver", as the user doesn't really need to think about whether PHP or libmongoc is doing the heavy lifting. The PHP extension is a black box for them.

My earlier explanation about libmongoc and PHP was more to provide you context, and explain why we (the PHP team) don't have much knowledge about the AWS mechanisms.